protocol/contracts/v2/templeLineOfCredit/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,962 @@
+        AccountData storage _accountData = allAccountsData[onBehalfOf];
+
+        // Get the outstanding debt for Stable
+        uint128 repayAmount = currentAccountDebt(
+            _debtTokenCache,
+            _accountData.debtCheckpoint,
+            _accountData.interestAccumulator,
+            true // use the rounded up amount
+        );
+        if (repayAmount == 0) revert CommonEventsAndErrors.ExpectedNonZero();
+        repayToken(_debtTokenCache, repayAmount, _accountData, msg.sender, onBehalfOf);
+    }
+
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                       LIQUIDATIONS                         */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @notice Liquidate one or more accounts which have exceeded the 
+     * maximum allowed LTV.
+     * The Temple collateral is seized, and the accounts debt wiped.
+     * @dev If one of the accounts in the batch hasn't exceeded the max LTV
+     * then no action is performed for that account.
+     */
+    function batchLiquidate(
+        address[] memory accounts
+    ) external override returns (
+        uint256 totalCollateralClaimed,
+        uint256 totalDebtWiped
+    ) {
+        LiquidationStatus memory _status;
+        DebtTokenCache memory _debtTokenCache = debtTokenCache();
+        address _account;
+        uint256 _numAccounts = accounts.length;
+        for (uint256 i; i < _numAccounts; ++i) {
+            _account = accounts[i];
+            _status = computeLiquidity(
+                allAccountsData[_account], 
+                _debtTokenCache, 
+                false
+            );
+
+            // Skip if this account is still under the maxLTV
+            if (_status.hasExceededMaxLtv) {
+                emit Liquidated(_account, _status.collateral, _status.collateralValue, _status.currentDebt);
+                totalCollateralClaimed += _status.collateral;
+                totalDebtWiped += _status.currentDebt;
+
+                // Clear the account data
+                delete allAccountsData[_account];
+            }
+        }
+
+        // burn the temple collateral by repaying to TRV. This will burn the equivalent dUSD debt too.
+        if (totalCollateralClaimed != 0) {
+            treasuryReservesVault.repayTemple(totalCollateralClaimed, address(tlcStrategy));
+            totalCollateral -= totalCollateralClaimed;
+        }
+
+        // Remove debt from the totals
+        repayTotalDebt(_debtTokenCache, totalDebtWiped.encodeUInt128());
+    }
+    
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                            ADMIN                           */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @notice Update the TLC Strategy contract, and Treasury Reserves Vault (TRV)
+     * @dev The TRV is granted access to spend DAI, in order to repay debt.
+     */
+    function setTlcStrategy(
```

Comment:
Since `trvDebtCeiling` might be changed, maybe it's needed to call `refreshInterestRates()`

Reply:
Done thanks